hal/targets/hal/TARGET_Freescale/TARGET_KSDK2_MCUS/TARGET_K64F/storage_driver.c

```diff
@@ -463,17 +575,18 @@ static inline void setup8ByteWrite(uint64_t addr, const void *data)
 #endif /* ifdef USING_KSDK2 */
 }
 
+__RAMFUNC
 static inline void setupProgramSection(uint64_t addr, const void *data, size_t cnt)
 {
 #ifdef USING_KSDK2
     static const uintptr_t FlexRAMBase = FSL_FEATURE_FLASH_FLEX_RAM_START_ADDRESS;
-    memcpy((void *)FlexRAMBase, (const uint8_t *)data, cnt);
+    MEMCPY((void *)FlexRAMBase, (const uint8_t *)data, cnt);
```

Comment:
I'm leaning towards this approach of copying code into a RAM buffer and executing it.
But the following issues still remain:
* How can we determine the size of the code to be copied? The way this is done in the KSDK driver (mentioned above by @c1728p9 is to depend on the address of the following function). That's not safe. I wish to avoid hand-coded assembly.
* Is there any way our 'non-const-array' holding code can land in a non-executable region of memory?